tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -393,6 +453,205 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+			}()
+
+			policyNames = append(policyNames, policyName)
+		}
+
+		if supervisorCluster {
+			assignPolicyToWcpNamespace(client, ctx, namespace, policyNames)
+		} else if guestCluster {
+			_, svNamespace := getSvcClientAndNamespace()
+			assignPolicyToWcpNamespace(client, ctx, svNamespace, policyNames)
+		}
+
+		for _, policyName := range policyNames {
+			if vanillaCluster {
+				ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+				scParameters[scParamStoragePolicyName] = policyName
+				storageclass, pvclaim, err = createPVCAndStorageClass(client,
+					namespace, nil, nil, "", nil, "", false, "")
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			} else if supervisorCluster {
+				ginkgo.By("CNS_TEST: Running for WCP setup")
+				profileID := e2eVSphere.GetSpbmPolicyID(policyName)
+				scParameters[scParamStoragePolicyID] = profileID
+				storageclass, err = client.StorageV1().StorageClasses().Get(ctx, policyName, metav1.GetOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				// create resource quota
+				createResourceQuota(client, namespace, rqLimit, policyName)
+				pvclaim, err = createPVC(client, namespace, nil, "", storageclass, "")
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			} else {
+				ginkgo.By("CNS_TEST: Running for GC setup")
+				createResourceQuota(client, namespace, rqLimit, policyName)
+				scParameters[svStorageClassName] = policyName
+				storageclass, err = client.StorageV1().StorageClasses().Get(ctx, policyName, metav1.GetOptions{})
+				if !apierrors.IsNotFound(err) {
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+				pvclaim, err = createPVC(client, namespace, nil, "", storageclass, "")
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+			pvcs = append(pvcs, pvclaim)
+			pvclaims2d = append(pvclaims2d, []*v1.PersistentVolumeClaim{pvclaim})
+			scs = append(scs, storageclass)
+		}
+
+		defer func() {
+			if vanillaCluster {
+				ginkgo.By("Delete the SCs created in step 2")
+				for _, sc := range scs {
+					err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+			}
+		}()
+
+		ginkgo.By("Verify the PVCs created in step 3 are bound")
+		pvs, err := fpv.WaitForPVClaimBoundPhase(client, pvcs, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+
+		volumeID := pvs[0].Spec.CSI.VolumeHandle
+		if guestCluster {
+			volumeID = getVolumeIDFromSupervisorCluster(pvs[0].Spec.CSI.VolumeHandle)
+			gomega.Expect(volumeID).NotTo(gomega.BeEmpty())
+		}
+		storagePolicyExists, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[0])
+		e2eVSphere.verifyVolumeCompliance(volumeID, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(storagePolicyExists).To(gomega.BeTrue(), "storage policy verification failed")
```

Comment:
we should verify this for each volume
also move `e2eVSphere.verifyVolumeCompliance(volumeID, true)` post the `Expect()` statements